src/Grid/GridHeader.js

```diff
@@ -104,44 +106,110 @@ class GridHeader extends React.Component {
+        column.Filter.Operator = this.state[this.state.activeFilter];
+        column.Filter.HasFilter = hasFilter;
         if(secondValue !== undefined){
-          row.Filter.Argument = [secondValue];
+          column.Filter.Argument = [secondValue];
         }
       }
     });
 
     this.state.dataSource.filter(this.state.rowsPerPage, this.state.page);
   }
 
+  handleKeyDown(event) {
+    if(this.state.keyState === 'Up' && event.ctrlKey) {
+      this.setState({ keyState: 'Down' });
+    } 
+  }
+
+  handleKeyUp(event) {
+    if(this.state.keyState === 'Down') {
+      this.setState({ keyState: 'Up' });
+    } 
+  }
+
+  componentDidMount() {
+    document.addEventListener('keydown', () => this.handleKeyDown(event));
+    document.addEventListener('keyup', () => this.handleKeyUp(event));
+  }
+
   sortHandler = (event, property) => {
+    if(this.state.keyState === 'Down'){
+      this.multiSort(event, property);
+    }
+    else{
+      this.singleSort(event, property);
+    }
+  }
+
+  multiSort = (event, property) => {
+    let sortOrder = this.state.sortOrder;
+    const array = Object.assign({}, this.state.dataSource);
+    
+    array.columns.forEach( (column, i) => {
+      if(column.Name === property){
+        if(column.SortOrder === -1) {
+          column.SortOrder = sortOrder;
+        }
+        else if(column.SortOrder !== -1 && column.SortDirection === 'Descending'){
+          column.SortOrder = -1;
+        }
+
+        const currentlySortedColumns = array.columns.filter(col => col.SortOrder > 0).sort((a, b) => a.SortOrder === b.SortOrder ? 0 : a.SortOrder > b.SortOrder );
+
+        for (let i = 0; i < currentlySortedColumns.length; i ++) { 
+          sortOrder = i + 2;
+
+          array.columns.forEach( (column, j) => {
+            if(column.Name === currentlySortedColumns[i].Name){
+              array.columns[j].SortOrder = (i + 1);
+            }
+          }); 
+        }
+
+        column.SortDirection = column.SortDirection === 'Ascending' ? 
+          'Descending' 
+          : 
+          column.SortDirection === 'Descending' ? 
+            'None' 
+            :
+            'Ascending';
+      }
+    });
+
+    this.setState({ sortOrder }, this.state.dataSource.sort(this.state.rowsPerPage, this.state.page) );
+  }
+
+  singleSort = (event, property) => {
     let sortOrder = 1;
 
-    this.state.dataSource.columns.forEach( row => {
-      if(row.Name === property){
-        if(row.SortDirection === 'Descending') {
+    this.state.dataSource.columns.forEach( column => {
+      if(column.Name === property){
+        if(column.SortDirection === 'Descending') {
           sortOrder = -1;
         }
 
-        row.SortOrder = sortOrder;
+        column.SortOrder = sortOrder;
         this.state.dataSource.sort(this.state.rowsPerPage, this.state.page);
-        row.SortDirection = row.SortDirection === 'Ascending' ? 
+        column.SortDirection = column.SortDirection === 'Ascending' ? 
```

Comment:
Please take a look at https://github.com/unosquare/tubular/blob/74d0eee2735dc0b966a4dca893974fa895414a75/src/js/tubular/directives/grid/tbGridCtrl.js#L434

Same validation is there and it's more readable there